src/sql/parts/dashboard/common/dashboardPage.component.ts

```diff
@@ -138,20 +138,29 @@ export abstract class DashboardPage extends Disposable implements OnDestroy {
 
 		let allTabs = dashboardHelper.filterConfigs(dashboardRegistry.tabs, this.dashboardService);
 
-		// Load always show tabs
-		let alwaysShowTabs = allTabs.filter(tab => tab.alwaysShow);
-		this.loadNewTabs(alwaysShowTabs);
+		// Load tab setting configs
+		this._tabSettingConfigs = this.dashboardService.getSettings<Array<TabSettingConfig>>([this.context, 'tabs'].join('.'));
 
-		// Load pinned tabs
-		this._pinnedTabs = this.dashboardService.getSettings<Array<PinConfig>>([this.context, 'tabs'].join('.'));
 		let pinnedDashboardTabs: IDashboardTab[] = [];
-		this._pinnedTabs.forEach(pinnedTab => {
-			let tab = allTabs.find(i => i.id === pinnedTab.tabId);
-			if (tab) {
-				pinnedDashboardTabs.push(tab);
+		let alwaysShowTabs = allTabs.filter(tab => tab.alwaysShow);
```

Comment:
I'm quite sure the logic is right here.  I think the "always show" are really "pinned by default after install".  Meaning that if a "alway show" tab is ever unpinned or closed it won't be shown again by default the next time the dashboard is displayed.  

I can't really tell from reading the code, but it doesn't appear to be implemented this way.  It looks like "always show" will, well, always show?

Reply:
There is no code to identify if the extension is freshly install since install extension only unzip the file and put it in the user folder. Instead, I implemented in such a way that it always show unless it is specified in the tab setting config. So the user can overwrite always show by unpinned the tab.  

---
In reply to: [173349906](https://github.com/Microsoft/sqlopsstudio/pull/879#discussion_r173349906) [](ancestors = 173349906)

Reply:
great, thanks!  Also, can you confirm that closing a tab will also unpin it if it is in a pinned state?  That seems to be normal pinning behavior as far I've seen.

Reply:
Chatted offline. Close action only close the tab. It doesn't unpin the tab

---
In reply to: [173520890](https://github.com/Microsoft/sqlopsstudio/pull/879#discussion_r173520890) [](ancestors = 173520890)